drivers/sensors/cirque_pinnacle.c

```diff
@@ -142,14 +156,19 @@ void ERA_WriteByte(uint16_t address, uint8_t data) {
     cirque_pinnacle_clear_flags();
 }
 
-void cirque_pinnacle_set_adc_attenuation(uint8_t adcGain) {
+bool cirque_pinnacle_set_adc_attenuation(uint8_t adcGain) {
     uint8_t adcconfig = 0x00;
 
     ERA_ReadBytes(EXTREG__TRACK_ADCCONFIG, &adcconfig, 1);
     adcconfig &= EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK;
-    adcconfig |= adcGain;
+    if (adcconfig == (adcGain & EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK)) {
+        return false;
+    }
+    adcconfig |= adcGain & EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK;
```

Comment:
I made a mistake with this mask.
Can you change to `#define EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK 0xC0`, then:
```suggestion
    adcGain &= EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK;
    if ((adcconfig & EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK) == adcGain) {
        return false;
    }
    adcconfig &= ~EXTREG__TRACK_ADCCONFIG__ADC_ATTENUATE_MASK;
    adcconfig |= adcGain;
```
Sorry for the confusion.

Reply:
Fixed in fixup commit